src/Components/Web/src/Forms/InputRadioGroup.cs

```diff
@@ -26,22 +26,35 @@ public class InputRadioGroup : ComponentBase
         /// </summary>
         [Parameter] public string? Name { get; set; }
 
+        [CascadingParameter] private InputRadioContext? CascadedContext { get; set; }
+
         /// <inheritdoc />
         protected override void OnParametersSet()
         {
-            GroupName = !string.IsNullOrEmpty(Name) ? Name : _defaultGroupName;
+            var changeEventCallback = EventCallback.Factory.CreateBinder<string?>(this, __value => CurrentValueAsString = __value, CurrentValueAsString);
+            var groupName = !string.IsNullOrEmpty(Name) ? Name : _defaultGroupName;
+
+            _context = new InputRadioContext(CascadedContext, groupName, CurrentValue, changeEventCallback);
         }
 
         /// <inheritdoc />
         protected override void BuildRenderTree(RenderTreeBuilder builder)
         {
-            Debug.Assert(GroupName != null);
-
-            builder.OpenComponent<CascadingValue<InputRadioGroup>>(0);
-            builder.AddAttribute(1, "IsFixed", true);
-            builder.AddAttribute(2, "Value", this);
-            builder.AddAttribute(3, "ChildContent", ChildContent);
+            Debug.Assert(_context != null);
+
+            builder.OpenElement(0, "div");
```

Comment:
Rendering a container element feels a bit surprising. I suppose it is necessary if we want to have a single place to put `CssClass`. However I'm unsure this will match up with people's use cases. Reasoning:

* Radio buttons are different from other form elements in that a single group of them might be visually distributed widely across the page. If we're limited to applying validation styling to their closest common ancestor, that might be a really large part of the page (maybe something bigger than the visible scroll area of the screen in extreme cases). You probably wouldn't want to highlight that whole area with a red box or shaded background, nor would you necessarily want to highlight all the radio buttons inside it, in case some of them are unrelated to this field.
* In the nested case from my example before ("Pick one color and one country") the two `InputRadioGroup` ancestor containers are exactly overlapping, so putting a validation class on one of them doesn't give the user any visual indication of which of the sets of radio buttons are invalid.

So I wonder if it would be safer, as a default, to apply the validation CSS class to all the `InputRadio` elements within the group, rather than to the group itself. That way the developer can (for example) set up the CSS to change just their text or circle colors to red, and not any unrelated radio buttons that happen to be nearby.

In the less common case where a developer does want to have a red box appear around the entire group, this would still be achievable using the existing methods for using the validation status of a single field to change a CSS class on an arbitrary element.

Additionally it would save us from having to pick a specific element type to use (`div` in this case) which people would then want to have a way to customize.

Reply:
My hope is this would be a pretty trivial change, since each of the `InputRadio` components already has access to the owning `InputRadioGroup` (via context) so it can just directly read and render the `CssClass` property from the group.

Reply:
I agree, this makes a lot of sense. I've gone ahead and made this change :)